### PR TITLE
Test hashing algorithms for speed

### PR DESCRIPTION
**Pull request by MatthewHambley**
There was some concern that the Adler32 checksum used in #38 was a pure python implementation and thus too slow. This didn't seem likely as comes from the `zlib` package and therefore, presumably, uses the `zlib` implementation.

A simple experiment was written to benchmark the various hashing algorithms. This confirms what we expected, specifically that the two data integrity checksums outperform the cryptographic hashes by 2 orders of magnitude.

### CODE REVIEW

File: Experimental/BenchmarkHashes/hashbench.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env python3
```
Comment (stevewardle) [inline Experimental/BenchmarkHashes/hashbench.py]: Perhaps we want a copyright header here (I note that I didn't put one in the existing `Experimental` test code, but I probably should have (not sure if we can justify adding them as part of this change?  Slightly out of scope but raising a PR just to modify them feels like hassle

File: Experimental/BenchmarkHashes/psykal_lite_mod.F90 [hunk 2]
```diff
@@ -0,0 +1,2669 @@
+!-----------------------------------------------------------------------------
```
Comment (stevewardle) [inline Experimental/BenchmarkHashes/psykal_lite_mod.F90]: I know this comes from another project but I think we should modify this copyright header in some way, or add a clarifying line at the end to make it clear that this is not applicable to the Fab project and is just from this example file's original implementation

File: Experimental/BenchmarkHashes/hashbench.py (hunk 1 above)
Comment (MatthewHambley) [reply Experimental/BenchmarkHashes/hashbench.py]: Done

File: Experimental/BenchmarkHashes/psykal_lite_mod.F90 (hunk 2 above)
Comment (MatthewHambley) [reply Experimental/BenchmarkHashes/psykal_lite_mod.F90]: And done.
I've copied the licence text from the donor project, as I should have done in the first place.

**Pull request merged**